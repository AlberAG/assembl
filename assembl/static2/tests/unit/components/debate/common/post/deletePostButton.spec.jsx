import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import DeletePostButton from '../../../../../../js/app/components/debate/common/deletePostButton';

describe('DeletePostButton component', () => {
  it('should render a delete post button', () => {
    const deletePostSpy = jest.fn();
    const props = {
      deletePost: deletePostSpy,
      postId: 'XYZ333',
      refetchQueries: [
        {
          query: 'dummyGraphqlQuery',
          variables: {
            lang: 'en'
          }
        }
      ]
    };
    const renderer = new ShallowRenderer();
    renderer.render(<DeletePostButton {...props} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});