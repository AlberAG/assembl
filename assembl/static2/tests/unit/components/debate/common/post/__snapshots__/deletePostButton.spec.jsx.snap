// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeletePostButton component should render a delete post button 1`] = `
<DeletePostButton
  deletePost={[Function]}
  postId="XYZ333"
  refetchQueries={
    Array [
      Object {
        "query": "dummyGraphqlQuery",
        "variables": Object {
          "lang": "en",
        },
      },
    ]
  }
/>
`;
